applications/visit/run.sh

```diff
@@ -0,0 +1,211 @@
+set -x
+
+echo "TACC: job $SLURM_JOB_ID execution at: `date`"
+
+########################################################
+##########     INTERACTIVE WRAPPER CONFIG     ##########
+########################################################
+
+# program and command line arguments run within xterm -e command
+XTERM_CMD="${_XTERM_CMD}"
+
+# Webhook callback url for job ready notification.
+# Notifications are sent to INTERACTIVE_WEBHOOK_URL i.e. https://3dem.org/webhooks/interactive/
+INTERACTIVE_WEBHOOK_URL="${_webhook_base_url}"
+
+########################################################
+########################################################
+########################################################
+
+# our node name
+NODE_HOSTNAME=`hostname -s`
+
+# HPC system target. Used as DCV host
+HPC_HOST=`hostname -d`
+
+echo "TACC: running on node $NODE_HOSTNAME on $HPC_HOST"
+
+TAP_FUNCTIONS="/share/doc/slurm/tap_functions"
+if [ -f ${TAP_FUNCTIONS} ]; then
+  . ${TAP_FUNCTIONS}
+else
+  echo "TACC:"
+  echo "TACC: ERROR - could not find TAP functions file: ${TAP_FUNCTIONS}"
+  echo "TACC: ERROR - Please submit a consulting ticket at the TACC user portal"
+  echo "TACC: ERROR - https://portal.tacc.utexas.edu/tacc-consulting/-/consult/tickets/create"
+  echo "TACC:"
+  echo "TACC: job $SLURM_JOB_ID execution finished at: $(date)"
+  exit 1
+fi
+
+# confirm DCV server is alive
+DCV_SERVER_UP=`systemctl is-active dcvserver`
+if [ $DCV_SERVER_UP == "active" ]; then
+  SERVER_TYPE="DCV"
+else
+  echo "TACC:"
+  echo "TACC: ERROR - could not confirm dcvserver active, systemctl returned '$DCV_SERVER_UP'"
+  SERVER_TYPE="VNC"
+fi
+
+# if X0 socket exists, then DCV will use a higher X display number and ruin our day
+# therefore, cowardly bail out and appeal to an admin to fix the problem
+if [ -f /tmp/.X11-unix/X0 ]; then
+  echo "TACC:"
+  echo "TACC: ERROR - X0 socket already exists. DCV script will fail."
+  echo "TACC: ERROR - Please submit a consulting ticket at the TACC user portal"
+  echo "TACC: ERROR - https://portal.tacc.utexas.edu/tacc-consulting/-/consult/tickets/create"
+  echo "TACC:"
+  SERVER_TYPE="VNC"
+fi
+
+# create an X startup file in /tmp
+# source xinitrc-common to ensure xterms can be made
+# then source the user's xstartup if it exists
+XSTARTUP="/tmp/dcv-startup-$USER"
+cat <<- EOF > $XSTARTUP
+#!/bin/sh
+
+unset SESSION_MANAGER
+unset DBUS_SESSION_BUS_ADDRESS
+. /etc/X11/xinit/xinitrc-common
+exec startxfce4
+EOF
+
+chmod a+rx $XSTARTUP
+
+if [ "x${SERVER_TYPE}" == "xDCV" ]; then
+  # create DCV session for this job
+  DCV_HANDLE="${TAPIS_JOB_ID}-session"
+  dcv create-session --owner ${TAPIS_JOB_OWNER} --init=$XSTARTUP $DCV_HANDLE
```

Comment:
The macros are a bit different for Tapis vs Agave:
https://tapis.readthedocs.io/en/latest/technical/jobs.html?highlight=_tapis#environment-variables

Can you adjust the TAPIS_JOB_OWNER and TAPIS_JOB_ID vars?